### ISSUE TITLE
Provide stable Docker images for latest tag

### ISSUE THREAD

**Issue by bbannier** [zeek/spicy#683]
Currently we build images for alpine-3.12, centos-8, and ubuntu-18 nightly. These images are tagged with both their version and `latest`, so that `latest` always pulls down the most recent nightly build. With the upcoming release in mind, we should change this scheme so that `latest` points to the most recent tag (release), while `nightly` can point to the latest nightly release.

Similarly, we should also provide stable binary artifacts for the latest tag. Binary artifacts are currently generated from Cirrus tasks and referenced in the docs. The model for these is probably to upload them as GH artifacts for release builds, and to update the docs to reference these artifacts instead.

**Comment by bbannier** [comment]
@sethhall independently brought up an interesting about our images: Right we provide a lot of choice to users instead of e.g., a simple `docker run -it zeekurity/spicy`. To enable that with the context of this issue we could e.g., provide simple distro-agnostic containers like `spicy:1.0.0` (maps to `1.0.0` tag), `spicy:latest` (maps to latest release) or `spicy:nightly` (maps to master state), or alternatively/in parallel move the distro into the image tag e.g., `spicy:1.0.0-ubuntu-20`.

**Comment by bbannier** [comment]
Moved the binary artifacts part of this issue into a new issue, #699.
